src/Util/AppVersion.m

```diff
@@ -0,0 +1,64 @@
+//
+//  AppVersion.m
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "AppVersion.h"
+
+@interface AppVersion ()
+
+@property (nonatomic) NSString *firstAppVersion;
+@property (nonatomic) NSString *lastAppVersion;
+@property (nonatomic) NSString *currentAppVersion;
+
+@end
+
+#pragma mark -
+
+@implementation AppVersion
+
++ (instancetype)instance {
+    static AppVersion *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        instance = [AppVersion new];
+        [instance configure];
+    });
+    return instance;
+}
+
+- (void)configure {
```

Comment:
Yes, at the risk of some redundancy, I wanted to include something that:

a) Was lightweight and could easily be made a dependency by any component in the system.
b) Is intended to be activated _very early_ in the app startup process.
c) Had certain specific side effects like logging.

I didn't want to touch something as important as migration logic, and since this logic is basically trivial, I wasn't too worried about the duplication.